data/store/mongo/mongo_data.go

```diff
@@ -0,0 +1,1042 @@
+	selector := bson.M{
+		"_active": true,
+		"_userId": id,
+		"type":    "cbg",
+		// we currently pull all so we can do in-range and avg in 1 pass, maybe bad idea
+		//"value": bson.M{"$gt": 3.9, "$lt": 10},
+		// NOTE: redundant query for migration of time field
+		"$or": bson.A{
+			bson.M{"time": bson.M{"$gte": startTime.Format(time.RFC3339Nano),
+				"$lte": endTime.Format(time.RFC3339Nano)}},
+			bson.M{"time": bson.M{"$gte": startTime,
+				"$lte": endTime}},
+		},
+	}
+
+	cursor, err := d.Find(ctx, selector)
+
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to get in-range values")
+	}
+
+	if err = cursor.All(ctx, &dataSets); err != nil {
+		return nil, errors.Wrap(err, "unable to decode data sets for user by id")
+	}
+
+	return dataSets, nil
+}
+
+func (d *DataRepository) GetLastUpdatedForUser(ctx context.Context, id string) (summary.UserLastUpdated, error) {
+	var status summary.UserLastUpdated
+
+	if ctx == nil {
+		return status, errors.New("context is missing")
+	}
+
+	if id == "" {
+		return status, errors.New("id is missing")
+	}
+
+	var dataSet []*continuous.Continuous
+	selector := bson.M{
+		"_active": true,
+		"_userId": id,
+		"type":    "cbg",
+	}
+
+	findOptions := options.Find()
+	findOptions.SetSort(bson.D{{Key: "time", Value: -1}})
+	findOptions.SetLimit(1)
+
+	cursor, err := d.Find(ctx, selector, findOptions)
+
+	if err != nil {
+		return status, errors.Wrap(err, "unable to get last cbg date")
+	}
+
+	if err = cursor.All(ctx, &dataSet); err != nil {
+		return status, errors.Wrap(err, "unable to decode last cbg date")
+	}
+
+	if dataSet != nil {
+		status.LastUpload, err = time.Parse(time.RFC3339Nano, *dataSet[0].CreatedTime)
+		if err != nil {
+			return status, errors.Wrap(err, "unable to parse latest CreatedTime")
+		}
+
+		status.LastData, err = time.Parse(time.RFC3339Nano, *dataSet[0].Time)
+		if err != nil {
+			return status, errors.Wrap(err, "unable to parse latest Time")
+		}
+	} else {
+		return status, errors.Wrap(err, "No cbg records found for user")
+	}
+
+	return status, nil
+}
+
+func (d *DataRepository) UserHasData(ctx context.Context, id string) (bool, error) {
+
+	if ctx == nil {
+		return false, errors.New("context is missing")
+	}
+
+	if id == "" {
+		return false, errors.New("id is missing")
+	}
+
+	var dataSet *data.DataSet
+	selector := bson.M{
+		"_userId": id,
+	}
+
+	err := d.FindOne(ctx, selector).Decode(&dataSet)
+
+	if err == mongo.ErrNoDocuments {
+		return false, nil
+	} else if err != nil {
+		return false, errors.Wrap(err, "unable to get data set")
+	}
+
+	return true, nil
+}
+
+func (d *DataRepository) GetFreshUsers(ctx context.Context, lastUpdated time.Time) ([]string, error) {
+	var userIDs []string
+
+	if ctx == nil {
+		return userIDs, errors.New("context is missing")
+	}
+
+	selector := bson.M{
+		"time":    bson.M{"$gte": lastUpdated.Format(time.RFC3339Nano)},
+		"type":    "cbg",
+		"_active": true,
+	}
+
+	result, err := d.Distinct(ctx, "_userId", selector)
```

Comment:
Is this an indexed query?

Reply:
no... its not :(
guess i forgot that one when last looking at distinct indexing issues.

The best query i can come up with currently that tricks the engine to use index without a fetch stage is:
```
db.deviceData.explain().distinct("_userId", {time: {$gte:"2001"}, type:"cbg", _active:true, _userId:{$ne: -111}})
```
which is slightly ugly, but probably safe? putting a more specific invalid userid match in, like null, doesnt trick it.

Reply:
perhaps this could also be applied to the previously-abandoned more specific query for DistinctUserIDs as well, which I removed all conditions from to get it to index.

Reply:
No... no its not.
I missed this one in my last pass at trying to get these indexed.
Adding a `"_userId": bson.M{"$ne": -1111}` condition to these queries tricks the distinct into being indexed while being a filtered query, and seems like the cleanest way to trick it.
I'll also go back and add this trick to the `DistinctUserIDs` query to make it only grab relevant users, as well.